app/routines/update_user_salesforce_info.rb

```diff
@@ -145,6 +145,40 @@ def call
     self
   end
 
+  def contacts
+    # The query below is not particularly fast, takes around a minute.  We could
+    # try to do something fancier, like only query contacts modified in the last day
+    # or keep track of when the SF data was last updated and use those timestamps
+    # to limit what data we pull from Salesforce (could have a global field in redis
+    # or could copy SF contact "LastModifiedAt" to a "sf_refreshed_at" field on each
+    # User record).
+    #
+    # Here's one example query as a starting point:
+    #   ...Contact.order("LastModifiedDate").where("LastModifiedDate >= #{1.day.ago.utc.iso8601}")
+    #
+    # TODO: Need to move the duplicated email detection code to SF before we change anything
+    #       If we use timestamps to limit results returned, need to be very careful
+    #       to avoid race conditions such as missing records that were modified
+    #       while we were querying SF
+    #       If we don't use timestamps, should load the contacts in chunks of 1,000 or 10,000
+    #       Or maybe try https://github.com/gooddata/salesforce_bulk_query
+
+    @contacts ||= OpenStax::Salesforce::Remote::Contact
```

Comment:
perhaps this is a candidate to do regularly in a background job, caching info in redis for fast access?

Anything taking a minute to query should be strongly considered for a background job :)

Reply:
I couldn't agree more! I cannot get wait for the chance to refactor this whole thing! 😅 

Same as above though, I am not the creator of this code and refactoring this is outside the scope of this PR. https://github.com/openstax/accounts/pull/837#discussion_r428296423

Reply:
Great suggestion 👍 👍 . I've been wanting to improve this code so now I think it's time. I created a new issue and will be adding your suggestions to it. Thanks!

Reply:
sorry...i clicked around too much, unresolved it, and then resolved it again as you had it.